pacta/email.go

```diff
@@ -0,0 +1,53 @@
+package pacta
+
+import (
+	"fmt"
+	"strings"
+	"unicode"
+	"unicode/utf8"
+
+	normalizer "github.com/dimuska139/go-email-normalizer"
+)
+
+var n = normalizer.NewNormalizer()
+
+func CanonizeEmail(email string) (string, error) {
+	// Performs basic validation, preventing obviously malformed email addresses, but
+	// not capturing the entirety of the RFC 5322 grammar.
+	split := strings.Split(email, "@")
+	if len(split) != 2 {
+		return "", fmt.Errorf("invalid email, wrong number of at-signs: %q", email)
+	}
+	if !isASCII(split[0]) || !utf8.ValidString(split[0]) {
```

Comment:
Hold up now: https://en.wikipedia.org/wiki/Email_address#Internationalization

Perfectly reasonable email addresses can have non-ASCII characters in them. Again, thinking of our European users, I'd expect umlauts and other things to appear in addresses, which are valid UTF-8 and valid email addresses by [RFC 6532](https://www.rfc-editor.org/rfc/rfc6532).

If internationalization was done via [punycode](https://en.wikipedia.org/wiki/Punycode), like domain names, then the ASCII check would be fine, but it's just full on Unicode

Reply:
Wait I'm confused - that spec seems to claim that it DOES have to be utf-8 encoded, no?

https://en.wikipedia.org/wiki/International_email#:~:text=The%20set%20of,fully%20internationalized%20email.

Reply:
VERY GOOD CATCH. Thank you!